services/base-service/spa/client/public/elements/pages/expert/app-expert.js

```diff
@@ -154,6 +154,36 @@ export default class AppExpert extends Mixin(LitElement)
     let websites = graphRoot.contactInfo?.filter(c => (!c['isPreferred'] || c['isPreferred'] === false) && c['rank'] === 20 && c.hasURL);
     websites.forEach(w => {
       if( !Array.isArray(w.hasURL) ) w.hasURL = [w.hasURL];
+
+      // create 'name' label with abbrev url if not present
+      w.hasURL.forEach(url => {
+        if( !url.name ) {
+          // remove http(s)://, www., and trailing slashes
+          url.name = url.url?.replace(/^(http:\/\/|https:\/\/)|www\.|\/*$/g, '');
+        }
+
+        try {
+          // also set custom icon depending on type of website
+          if( url['@type'].includes('URL_googlescholar') ) {
+            url.icon = 'fa-google-scholar';
+          } else if( url['@type'].includes('URL_researchgate') ) {
+            url.icon = 'fa-researchgate';
+          } else if( url['@type'].includes('URL_linkedin') ) {
+            url.icon = 'fa-linkedin';
+          } else if( url['@type'].includes('URL_twitter') ) {
+            url.icon = 'fa-x-twitter';
+          } else if( url['@type'].includes('URL_mendeley') ) {
+            url.icon = 'fa-mendeley';
+          } else if( url['@type'].includes('URL_rssfeed') ) {
+            url.icon = 'fa-square-rss';
+          } else if( url['@type'].includes('URL_figshare') ) {
+            url.icon = 'ai-figshare';
+          }
+        } catch(e) {
+          console.warn('error setting website icon', e);
```

Comment:
@UcDust we should just be overwriting a standard link icon right? This isn't an error or even a warning.

Reply:
@qjhart the message would be triggered if `url['@type']` is not a string or an array of strings. Wouldn't that be an error with the api structure for type if it's hit?